pkgs/native_toolchain_c/test/cbuilder/testfiles/add_objective_c/src/add.m

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#import <Foundation/Foundation.h>
+
+#if DEBUG
+#import <stdio.h>
+#endif
+
+NSInteger add(NSInteger a, NSInteger b) {
+#ifdef DEBUG
+  printf("Adding %ld and %ld.\n", (long)a, (long)b);
+#endif
+  return a + b;
```

Comment:
> Do I need to use something from the Apple foundation libraries to break a dylib without that flag? Any suggestions?

`NSInteger` is just a typedef of `long`. You could try converting the `NSInteger` to a `NSNumber` and back. `NSNumber` is a real class, so doing this will require ObjC specific syntax, which will break without the language flag.

```Objective-c
return [[NSNumber numberWithInteger:(a + b)] longValue];
```

Or you could just do the addition inside a block, which would also require ObjC specific syntax:

```Objective-c
return [^long() {
  return a + b;
}();
```

Reply:
Objective-C syntax also simply works without the `-x objective-c` as long as the file is named `.m`. So maybe we just leave out the `-x` flag for now.
If in the future we end up having to compile mixed sources and they have the right extensions that would simply work
If in the future we end up having mixed sources that do not have the right extensions, we might need change the `sources` param to a map with languages as a key or add a `sourcesCpp` and `sourcesObjectiveC`.

> NSNumber

Thanks! Added.